### ISSUE TITLE
Improve third party dependencies documentation?

### ISSUE THREAD

**Issue by cristianoccazinsp** [react-navigation/react-navigation#6243]
### Expected Behavior

It looks like both version 3 and 4 require ```react-native-gesture-handler``` and ```react-native-reanimated```. However, I've been using version 3 without react-native-reanimated for a long time now (I didn't even notice the new dependency after an upgrade) without issues, is it really needed? Why? The same applies for react-native-screens.

Additionally, there seems to be some inconsistencies between the docs. https://reactnavigation.org/docs/en/stack-navigator.html says only react-native-gesture-handler and react-native-screens are needed, while the main docs (https://reactnavigation.org/docs/en/getting-started.html) say something different. 

Lastly, https://reactnavigation.org/docs/en/stack-navigator.html adds another two dependencies (@react-navigation/core and react-navigation-stack). Is this in addition to react-navigation?

In short, to install react-navigation-stack we have a bunch of dependencies scattered all around the docs, with some of them not even being required but rather optional. Considering also that those third party libraries can have a significant impact on the final build size due to all the native dependencies, it would be a very good idea to write a brief description about what they are used for and what are the drawbacks of not installing them.

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 3.12.1 and 4.0
| react-native     | 0.60.5

**Comment by satya164** [comment]
`react-native-reanimated` isn't used for stack, however it's used for both drawer and tabs. This is why I added them in the getting started documentation.

> The same applies for react-native-screens

screens is a peer dependency of stack and is imported in the package. Bundling will fail if the package isn't installed.

> Lastly, reactnavigation.org/docs/en/stack-navigator.html adds another two dependencies (@react-navigation/core and react-navigation-stack). Is this in addition to react-navigation?

core is already installed with react-navigation package. I had listed it in stack's page as well because it's peer dependency. I have replaced it with `react-navigation` now to reduce confusion.

> In short, to install react-navigation-stack we have a bunch of dependencies scattered all around the docs, with some of them not even being required but rather optional

All of the react-navigation-stack's dependencies are mentioned in stack navigator's page. Can you point out which dependencies are missing and scattered to other places?

All of the dependencies listed on stack's page are required for stack to work. If there is actually any optional dependencies listed on that page, please let me know.

**Comment by cristianoccazinsp** [comment]
It would be good for a note about reanimated then. As a stack only user, I
would definitely not install it to reduce the build size, but looking at
the docs it seems like I need it.

About react-screens, do we then need to install it manually it or not? I
swear I've never installed it and it it's working fine.

By docs scattered, I meant mostly that you seem to need to hunt down
dependencies from both getting started and the stack navigation docs.

Hope that helps, it's really not a big deal, but shocked me a bit when
looking how to migrate to v4.

**Comment by satya164** [comment]
> About react-screens, do we then need to install it manually it or not? I
swear I've never installed it and it it's working fine.

It's imported in stack. Metro won't bundle if it's not there. You do need to install it.

> I meant mostly that you seem to need to hunt down
dependencies from both getting started and the stack navigation docs.

All of the deps of stack are in stack's page. You don't need to check both getting started and stack's page. I just added them in getting started, but seems they are causing confusion, so I have removed them.
